src/infrastructure/galleryItem.repository.ts

```diff
@@ -9,7 +9,10 @@ export class GalleryItemRepository {
     return newItem;
   }
 
-  async getAll(): Promise<IGalleryItem[]> {
+  async getAll(limit?: number): Promise<IGalleryItem[]> {
+    if (limit) {
+      return await GalleryItemModel.find().limit(limit);
+    }
     return await GalleryItemModel.find();
```

Comment:
Instead of repeating the same statement twice, with and without `limit`, you could benefit from using Query Chaining like this. These chains are only activated when the query is `awaited`. So the query is only executed once `return await query` is reached.

```suggestion
  async getAll(limit?: number): Promise<IGalleryItem[]> {
    let query = GalleryItemModel.find();
    if (limit) {
      query = query.limit(limit);
    }
    return await query;
```